### DISCUSSION TITLE
LoRa SX1278, system halted after input LoRaSend command

### DISCUSSION THREAD

**Discussion by pcsl999** [arendst/Tasmota#21627] [Support request]
GPIO21 SPI MISO
GPIO23 SPI MOSI
GPIO22 SPI CLK
GPIO19 LoRa CS
GPIO15 LoRa Rst
GPIO18 LoRa GPIO0

00:00:00.002 HDW: ESP32-D0WD-V3 v3.1 
00:00:00.080 UFS: FlashFS mounted with 300 kB free
00:00:00.110 CFG: Loaded from File, Count 45
00:00:00.116 SPI: Bus1 using GPIO22(CLK), GPIO23(MOSI) and GPIO21(MISO)
00:00:00.190 BRY: Berry initialized, RAM used 3697 bytes
00:00:00.207 Project IOTResearch - Lora-280598-1432 Version 14.1.0.1(tasmota32)-3_0_0(2024-06-14T05:00:39)
00:00:00.456 LOR: SX127x initialized

06:46:30.344 CMD: LoRaConfig
06:46:30.353 MQT: stat/IOTResearch_280598/RESULT = {"LoRaConfig":{"Frequency":434.0,"Bandwidth":125.0,"SpreadingFactor":12,"CodingRate4":5,"SyncWord":18,"OutputPower":8,"PreambleLength":128,"CurrentLimit":60.0,"ImplicitHeader":0,"CrcBytes":2}}

06:46:16.967 CMD: LoRaSend
06:46:16.974 MQT: stat/IOTResearch_280598/RESULT = {"Command":"Error"}

**Input LoRaSend 0123456789, system will halt.**

**Comment by sfromis** [comment]
Doing the same test with a board based on SX1276, I saw no failure, just the data received on the other end. Usually "halt" would mean a crash, where the RestartReason might be "interesting".

**Comment by arendst** [comment]
No problems here too:

```
10:29:18.755-022 CMD: loraconfig
10:29:18.757-022 SRC: Serial
10:29:18.760-022 CMD: Grp 0, Cmd 'LORACONFIG', Idx 1, Len 0, Pld -99, Data ''
10:29:18.771-019 MQT: stat/wemos4/RESULT = {"LoRaConfig":{"Frequency":868.1,"Bandwidth":125.0,"SpreadingFactor":9,"CodingRate4":5,"SyncWord":52,"OutputPower":10,"PreambleLength":8,"CurrentLimit":60.0,"ImplicitHeader":0,"CrcBytes":2}}
10:29:31.472-022 HTP: Configure Module
10:29:58.821-022 CMD: lorasend
10:29:58.823-022 SRC: Serial
10:29:58.826-022 CMD: Grp 0, Cmd 'LORASEND', Idx 1, Len 0, Pld -99, Data ''
10:29:58.835-018 MQT: stat/wemos4/RESULT = {"Command":"Error"}
10:30:04.071-022 CMD: lorasend 0123456789
10:30:04.073-022 SRC: Serial
10:30:04.076-022 CMD: Grp 0, Cmd 'LORASEND', Idx 1, Len 10, Pld 342391, Data '0123456789'
10:30:04.237-018 MQT: stat/wemos4/RESULT = {"LoRaSend":"Done"}
```

**Comment by pcsl999** [comment]
It seem SX1278 issue. SX1276 are no more stock.
We can order LLCC68, SX1268.

22:27:44.915 HTP: Tools
22:27:45.573 HTP: Console
22:27:57.266 CMD: LoRaConfig
22:27:57.268 SRC: WebConsole from 192.168.10.50
22:27:57.269 CMD: Grp 0, Cmd 'LORACONFIG', Idx 1, Len 0, Pld -99, Data ''
22:27:57.278 MQT: stat/IOTResearch_280598/RESULT = {"LoRaConfig":{"Frequency":434.0,"Bandwidth":125.0,"SpreadingFactor":9,"CodingRate4":5,"SyncWord":52,"OutputPower":10,"PreambleLength":8,"CurrentLimit":60.0,"ImplicitHeader":0,"CrcBytes":2}}
22:28:13.707 CMD: LoRaSend 
22:28:13.708 SRC: WebConsole from 192.168.10.50
22:28:13.709 CMD: Grp 0, Cmd 'LORASEND', Idx 1, Len 0, Pld -99, Data ''
22:28:13.715 MQT: stat/IOTResearch_280598/RESULT = {"Command":"Error"}
22:28:25.346 CMD: LoRaSend 0123456789
22:28:25.347 SRC: WebConsole from 192.168.10.50
22:28:25.348 CMD: Grp 0, Cmd 'LORASEND', Idx 1, Len 10, Pld 342391, Data '0123456789'
22:28:25.362 DBG: Len 10, Send 30313233343536373839

**Reply by sfromis** [reply]
Likely a typo, but the gpio config you posted does not have a `LoRa DIO0` pin. Copy/paste of config data can be recommended, instead of replicating by typing.

**Comment by pcsl999** [comment]
SX1278TR4-Y pins RST, GPIO0, CSN, GOIP02, GPIO01, SO, SCLK, SI, GND, VCC, GPIO03, GPIO04, GPIO05, XOEN, GND, ANT, GND

ESP32 GPIO21 SPI MISO > BOARD SO
ESP32 GPIO23 SPI MOSI > BOARD SI
ESP32 GPIO22 SPI CLK > BOARD SCLK
ESP32 GPIO19 LoRa CS > BOARD CSN
ESP32 GPIO15 LoRa RST > BOARD RST
ESP32 GPIO18 LoRa DIO0 > BOARD GPIO0
